--- a//mnt/e/RTL_Optimizer-20260322T083825Z-3-001/RTL_Optimizer/rtl/clockGate.v
+++ b/clockGate_optimized.v
@@ -1,4 +1,5 @@
 module top(clk,ctrl,inp1,inp2,out);

+
	// ---------------------------------------------------------------------------
	// POWER OPTIMIZATION: AI-Injected Sequential Enable Logic
	// Target Register: r2
	// ---------------------------------------------------------------------------
	wire seq_enable_1;
	assign seq_enable_1 = (!ctrl);
	// ---------------------------------------------------------------------------

 	input clk,ctrl;

 	input [7:0] inp1,inp2;

 	output [7:0] out;

@@ -6,7 +7,7 @@
 	reg [7:0] r2;

 	always@(posedge clk)begin

 		r1 <= ctrl;

-		r2 <= inp1;

+		if (seq_enable_1) r2 <= inp1; 

 	end

 	assign out = r1 ? inp2 : r2;

 endmodule
